client/my-sites/site-settings/redirect-to/redirect-to.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpack = ( WrappedComponent, redirectUrl ) => {
+	class RedirectNonJetpack extends Component {
+		static propTypes = {
+			redirectUrl: PropTypes.string,
+			// Connected props
+			siteIsAtomic: PropTypes.bool,
+			siteIsJetpack: PropTypes.bool,
+			siteSlug: PropTypes.string
+		}
+
+		componentDidMount() {
+			this.verifySiteIsJetpackOrAtomic();
+		}
+
+		componentDidUpdate() {
+			this.verifySiteIsJetpackOrAtomic();
+		}
+
+		verifySiteIsJetpackOrAtomic() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirectTo();
+			}
+		}
+
+		redirectTo = () => {
```

Comment:
Nit: This strikes me as a strange name for a function that performs some navigation, both here and in the wrapped components. It sounds to me more like a property where you'd expect to find a path.